flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/functions/InternalWindowFunction.java

```diff
@@ -46,5 +46,5 @@
 	 *
 	 * @throws Exception The function may throw exceptions to fail the program and trigger recovery.
 	 */
-	public abstract void apply(KEY key, W window, IN input, Collector<OUT> out) throws Exception;
+	public abstract void process(KEY key, W window, IN input, Collector<OUT> out) throws Exception;
```

Comment:
Thanks for the feedback @manuzhang ! Definitely it has to be extended, otherwise there is no difference between WF and PWF. 
By the way, I added Scala support to PWF in my last commit.